### PR TITLE
Add snapshot test for raw_data and README

### PR DESCRIPTION
**Pull request by fridgepoet**
**What this PR does / why we need it**:
Based on the snapshot tests added during Elasticsearch's migration to a backend datasource. 

This creates snapshot tests on the backend to ensure that during the [migration to a backend datasource](https://github.com/grafana/opensearch-datasource/issues/192), we will not break request and response processing. Snapshot tests for newly supported query types should be added across the query migration. 

This also creates a README document with some presentation of the snapshot tests and how to add more.

There are two test types:
  * Test_query_type_request: Given query input from the data source frontend in Grafana, validates the request made to OpenSearch
  * Test_query_type_response: Given the response from OpenSearch, validates the snapshot of a data frame generated by the data source at the time of writing the test

**Which issue(s) this PR fixes**:

Contributes to https://github.com/grafana/opensearch-datasource/issues/255

**Special notes for your reviewer**:
This PR is based on https://github.com/grafana/grafana/pull/59312.

### CODE REVIEW

File: pkg/opensearch/snapshot_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#### To add a response test:
+1. To intercept the response from OpenSearch, set `debugEnabled = true` on the `baseClientImpl`. We'll use some existing code which reads the response body. There's a [method](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L458) that can be called. 
+2. Access an instance of Grafana with this data source and execute a query type for which we are adding a new test. You may want to limit the response by limiting the time range or setting a smaller size. 
+3. Set a breakpoint right after the body is read [here for Lucene](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L364) and [here for PPL](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L585). The `tmpBytes` JSON is the response from OpenSearch.
+4. Create a new empty response_from_opensearch file in `testdata` with the new query type, e.g.: `lucene_logs.response_from_opensearch.json`
+5. Copy the `tmpBytes` JSON into this file.
```
Comment (kevinwcyu) [inline pkg/opensearch/snapshot_tests/README.md:31]: it was difficult for me to copy the response while debugging with vscode. the response was truncated in the debugger so i ended up just printing it out with `backend.Logger.Debug("tmpBytes", "string(tmpBytes)", string(tmpBytes))` and then running it through `JSON.parse` with javascript.

am i missing an easier way to copy this? or is it just difficult with vscode?

File: pkg/opensearch/snapshot_tests/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			"pplEnabled":true,
+			"version":"2.3.0",
+			"timeField":"timestamp",
+			"interval":"Daily",
+			"timeInterval":"1s",
+			"maxConcurrentShardRequests":42
+		}`),
+		Database: "[testdb-]YYYY.MM.DD",
+		URL:      "http://localhost:9200",
+	}
```
Comment (kevinwcyu) [inline pkg/opensearch/snapshot_tests/helpers_test.go:52]: should we include things like the log level field and log message field in the settings?

File: pkg/opensearch/snapshot_tests/testdata/lucene_raw_data.query_input.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/* cSpell:disable */
```
Comment (kevinwcyu) [inline pkg/opensearch/snapshot_tests/testdata/lucene_raw_data.query_input.json:1]: looks like this is causing the tests to fail.

you can add `"**/testdata/*.json"` to the `ignorePaths` in `cspell.config.json`

**Review by kevinwcyu — APPROVED** [review]
nice! i got it working with ppl logs queries.

File: pkg/opensearch/snapshot_tests/README.md (hunk 1 above)
Comment (fridgepoet) [reply pkg/opensearch/snapshot_tests/README.md:31]: Darn! I was able to copy it with my Goland. Maybe it wasn't as long. Do you want me to try with VSCode?
Another idea is maybe you can query the OpenSearch dashboard directly? 
Did limiting the time range help or was setting a smaller size possible?

File: pkg/opensearch/snapshot_tests/helpers_test.go (hunk 2 above)
Comment (fridgepoet) [reply pkg/opensearch/snapshot_tests/helpers_test.go:52]: Good point, looks like I added those fields with the Lucene logs migration. I will add it in the upcoming snapshot test PR for Lucene logs

File: pkg/opensearch/snapshot_tests/testdata/lucene_raw_data.query_input.json (hunk 3 above)
Comment (fridgepoet) [reply pkg/opensearch/snapshot_tests/testdata/lucene_raw_data.query_input.json]: thanks!!

**Review by idastambuk — APPROVED** [review]

File: pkg/opensearch/snapshot_tests/README.md (hunk 1 above)
Comment (kevinwcyu) [reply pkg/opensearch/snapshot_tests/README.md:31]: i can only limit it to 3 fields and a single row.

i think there's some character limit in the debugger or some other setting that limits it.

**Pull request merged**